V7.1 update (#101)
iExecBlockchainComputing/documentation: for-developers/confidential-computing/requester-secrets.md

# Use requester secrets

{% hint style="success" %}
**Prerequisites**

* [Docker](https://docs.docker.com/install/) 17.05 or higher on the daemon and client.
* [Nodejs](https://nodejs.org) 14.0.0 or higher.
* [iExec SDK](https://www.npmjs.com/package/iexec) 7.2.0 or higher.
* Familiarity with the basic concepts of [Intel® SGX](intel-sgx-technology.md#intel-r-software-guard-extension-intel-r-sgx) and [SCONE](intel-sgx-technology.md#scone-framework) framework.
{% endhint %}

{% hint style="warning" %}
Please make sure you have already checked the [Quickstart](../your-first-app.md), [Your first application](../your-first-app.md) and [Build trusted applications](create-your-first-sgx-app.md) tutorials before learning how to use requester secrets.
{% endhint %}

Trusted Execution Environments offer a huge advantage from a security perspective. They guarantee that the behavior of execution does not change even when launched on an untrusted remote machine. The data inside this type of environment is also protected, which allows its monetization while preventing leakage.

> With iExec, it is possible to securely to associate an application developer secret to the runtime of an application.

Author: missed this one, thank you @jbern0rd 
what do you think of this?
```suggestion
With iExec, it is possible to securely consume requester-provided secrets in the application
```

iexec requester check-secret my-key --chain bellecour
```

```sh
# restore the default configuration in chain.json
sed -i 's|"bellecour": { "sms": "https://v7.sms.debug-tee-services.bellecour.iex.ec" },|"bellecour": {},|g' chain.json
```

### Run the TEE app

Specify the tag `--tag tee` in `iexec app run` command to run a tee app with an app developer secret.

One last thing, in order to run a **TEE-debug** app you will also need to select a debug workerpool, use the debug workerpool `v7-debug.main.pools.iexec.eth`.

You are now ready to run the app with requester secrets.

```sh
iexec app run <appAddress> \
  --tag tee \
  --workerpool v7-debug.main.pools.iexec.eth \
  --secret 1=my-namespace \
  --secret 2=my-key \
  --watch \
  --chain bellecour
```

{% hint style="info" %}
The option `--secret <secretMapping...>` allow the requester to provision any number of secrets with the mapping syntax `<key>=<name>`.

example:

```sh
--secret 1=foo 3=bar
```

* the secret named `foo` will be available in `IEXEC_REQUESTER_SECRET_1`
* the secret named `bar` will be available in `IEXEC_REQUESTER_SECRET_3`
* `IEXEC_REQUESTER_SECRET_2` will be skipped
{% endhint %}

## Next step?

> Thanks to the explained confidential computing workflow, it is possible to use consume requester secrets with a trusted application.

Author: fixed a00bac52dc76c2b15044a7796daf5dbdea849d51
